core/src/main/java/com/radixdlt/ledger/StateComputerLedger.java

```diff
@@ -154,6 +154,7 @@ void commit(
   private final Metrics metrics;
   private final LedgerAccumulator accumulator;
   private final LedgerAccumulatorVerifier verifier;
+  // TODO(resolve during review): Is it still needed? Was it ever needed?
   private final Object lock = new Object();
```

Comment:
_(a note from the author)_

Actually @malexandru-rdx brought my attention to this: apparently all my efforts to reduce "locking on too-wide scopes" on Rust side are killed by this lock on Java-side. Does anyone know a specific purpose of this extra lock here? Was there some weird effect going on, or is this just someone being too eager? Or is this a remnant from the days when the "state manager" was happening only on Java side?

Based on https://github.com/radixdlt/babylon-node/blame/29c163a2562b30d52e7db2a43e969a3847b042f5/radixdlt-core/radixdlt/src/main/java/com/radixdlt/ledger/StateComputerLedger.java and https://github.com/radixdlt/babylon-node/commit/e355b3ddebdf487b39da448b963a3ed1fdd5979e I suspect it is the "old sync from java days" case, but I need confirmation before I delete this.

Reply:
I would still merge this PR first before figuring out this lock.
Definitely the lock can be split between `prepare` + `commit` and `mempooladd` + `getTransactionsForProposal` and get (some of?) the benefits of your refactors.
I'm wondering if there is any reason not to remove it completely.